todolist/persistence/src/main/scala/runtime/TodoListRepositoryHandler.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package examples.todolist.persistence.runtime
+
+import cats.Monad
+import cats.effect.Bracket
+import doobie.implicits._
+import doobie.util.transactor.Transactor
+import examples.todolist.TodoList
+import examples.todolist.persistence.TodoListRepository
+
+class TodoListRepositoryHandler[F[_]: Monad](implicit T: Transactor[F], ev: Bracket[F, Throwable])
+    extends TodoListRepository[F] {
```

Comment:
For all these repositories, I'd use the free type `ConnectionIO` for the response. In that way, the algebra that use the repositories will receive the transaction and could combine freely the operations in a single transaction if needed. It also keeps the code simpler

```scala
class TodoListRepositoryHandler extends TodoListRepository[ConnectionIO] {

  import examples.todolist.persistence.runtime.queries.TodoListQueries._

  def insert(item: TodoList): F[Option[TodoList]] =
    insertQuery(item)
      .withUniqueGeneratedKeys[Int]("id")
      .flatMap(getQuery(_).option)

//...
```

Reply:
Great idea!  I'll give that a try.

Reply:
hey @fedefernandez, which library does the `ConnectionIO` type come from?  When I tried to use it in this refactor my editor said that it couldn't find the type in any of the existing imports and it didn't suggest any imports for me that made any sense (I tried adding `doobie.syntax._`) to no avail.

Reply:
It's `doobie.ConnectionIO`. It's a type alias of a type alias, which may have confused your IDE.

Reply:
🤦  lol that definitely might explain it.  Shoulda trusted my gut and kept reading the doobie docs.  Thanks Chris!

Reply:
This is done